src/communication/network_interface_cl.c

```diff
@@ -1798,6 +1902,47 @@ heap_reclaim_addresses (const HFID * hfid)
 #endif /* !CS_MODE */
 }
 
+/*
+ * heap_get_maxslotted_reclength -
+ *
+ * return:
+ *
+ *   maxslotted_reclength(out):
+ *
+ * NOTE:
+ */
+int
+heap_get_maxslotted_reclength (int &maxslotted_reclength)
+{
+#if defined(CS_MODE)
+  int req_error, success = ER_FAILED;
+  OR_ALIGNED_BUF (OR_INT_SIZE) a_reply;
+  char *reply;
+
+  reply = OR_ALIGNED_BUF_START (a_reply);
+  req_error =
+    net_client_request (NET_SERVER_HEAP_GET_MAXSLOTTED_RECLENGTH, NULL, 0, reply,
+			OR_ALIGNED_BUF_SIZE (a_reply), NULL, 0, NULL, 0);
+  if (!req_error)
+    {
+      or_unpack_int (reply, &maxslotted_reclength);
+      success = NO_ERROR;
+    }
+
+  return success;
+#else /* CS_MODE */
+  int success = ER_FAILED;
+
+  THREAD_ENTRY *thread_p = enter_server ();
+
+  success = xheap_get_maxslotted_reclength (maxslotted_reclength);
+
+  exit_server (*thread_p);
+
+  return success;
```

Comment:
Does this interface need to support non-CS-mode?
```suggestion
```

Reply:
Yes, it doesn't need. I just add like other function in this file for the futher usages.